dask/array/core.py

```diff
@@ -1416,8 +1416,15 @@ def elemwise(op, *args, **kwargs):
                    for arg in args)
     expr_inds = tuple(range(out_ndim))[::-1]
 
-    arrays = [arg for arg in args if isinstance(arg, Array)]
-    other = [(i, arg) for i, arg in enumerate(args) if not isinstance(arg, Array)]
+    arrays = []
+    other = []
+    for i, arg in enumerate(args):
+        if isinstance(arg, np.ndarray):
+            raise NotImplementedError
+        elif isinstance(arg, Array):
+            arrays.append(arg)
+        else:
+            other.append((i, arg))
```

Comment:
I'm -1 on changes like this.  I personally find the meaning of the list comprehensions to be more evident.  However I recognize that this is subjective; I wouldn't object to new code being written this way but prefer it if we don't go out of our way to rewrite existing code.

I suggest adding an extra line ahead of the two list comprehensions with the following check:

```python
if any(isinstance(arg, np.ndarray) for arg in args):
    raise NotImplemented("Dask.array operations only work on dask arrays, not numpy arrays.")
```

This isolates the type checking code from splitting code.